File: src/routes/ussd.js
const express = require('express');
const router = express.Router();

const UssdMenu = require('ussd-menu-builder');
const menu = new UssdMenu();

// USSD routes
router.get('/', (req, res) => {
    // res.render('ussd', res.locals.commonData);
    res.render('ussd.jade');
});

router.post('/', (req, res) => {
    menu.state('registration', {
        run: () =>{
            menu.con('Welcome to the registration system. \n Enter your username');
        },
        next: {
            '*[a-zA-Z]+': 'registration.name'
        }
    });
    
    menu.state('registration.name', {
        run: () => {
            let name = menu.val;
            let session = getSession(menu.args.sessionId);
            session.set('name', name);
            menu.con('Enter your email');
        },
        next: {
            '*\\w+@\\w+\\.\\w+': 'registration.email'
        }
    });

    menu.state('registration.email', {
        run: () => {
            let email = menu.val;
            let session = getSession(menu.args.sessionId);
            session.set('email', email);
            menu.end("Congratulations. You have successfully been registered into the system. \n You'll be hearing from us soon.");
        }
    });
    return menu;
});


// router.post('/', (req, res) => {
//     // Read variables sent via POST from the AT SDK
//     const { sessionId, serviceCode, phoneNumber, text } = req.body;

//     // split the various levels of the user input
//     const level = req.body.text.split('*');

//     let response = '';
//     if (!!text) {
//         if (text == '') {
//             // This is the first request
//             response = `CON Welcome to the registration portal. 
//                             \n Please enter your desired username`;
//         } else if ( !!level[0] && level[0] !== "" && !level[1] ) {
//             // Business logic for first level response
//             response = `CON Hi ${level[0]}, enter your email.`;
//         } else if ( !!level[1] && level[1] !== "" && !level[3] ) {
//             // Business logic for second level response
//             const data = [
//                 {"username": level[0]},
//                 {"email": level[1]}
//             ];
//             // This is the terminal request.
//             response = `END Thank you ${level[0]} for registering.
//                             \n We will keep you updated.`;
//         } 

//         // Print the response onto the page so that the AT SDK can read it
//         res.set('Content-Type: text/plain');
//         res.send(response);
//         // DONE!!!
//     }     
// });

module.exports = router;

File: Task2/ussd.py
import africastalking
from flask import Flask, request

# initialize
app = Flask(__name__)
username = "sandbox"
 
apikey = "2eb97013ea7b1ee3da4b2f902bed7392f4f6728dedd1831f2ad40fc6e1f1a156" 

africastalking.initialize(username, apikey)
sms = africastalking.SMS


#  route 
@app.route('/')
def index():
    return ""


@app.route('/ussd', methods = ['GET', 'POST'])
def ussd():
    text         = request.values.get("text", "default")
    sessionId   = request.values.get("sessionId", None)
    serviceCode  = request.values.get("serviceCode", None)
    phoneNumber = request.values.get("phoneNumber", None)
     
    if text == '':
        response = "Please enter your name \n"
        response += "Please do not enter @ in your username \n"

    elif ('@' not in text) :
        response = "Please enter your email address \n"

    else:
        response = "END You will receive an sms confirmation shortly"
        sms.send("You have successfully registered.", [phoneNumber])
    return response

if __name__ == '__main__':
    app.run()
